packages/components-library/src/components/filters/containers/FilterContainer.vue

```diff
@@ -209,7 +209,20 @@ export default {
        * @property {Object} - Filter name/value
        * @event input
        */
-      this.$emit('input', { ...this.value, [name]: value })
+
+      // clear an empty checkbox and range filter
+      if (Array.isArray(value)) {
+        if (value.length === 0) {
+          value = undefined
+        } else if (value.every(item => item === null)) {
+          value = undefined
+        }
+      }
+
+      const newSelection = { ...this.value, [name]: value }
+      // remove all undefined properties
+      Object.keys(newSelection).forEach(key => newSelection[key] === undefined && delete newSelection[key])
```

Comment:
why` a === b && do action` instead of if `( a === b ) { do action } `
or `keys(ns).filter(somefilter).forEach(doAction)`